qiskit/primitives/primitive_job.py

```diff
@@ -27,14 +27,16 @@ class PrimitiveJob(BasePrimitiveJob[ResultT, JobStatus]):
     Primitive job class for the reference implementations of Primitives.
     """
 
-    def __init__(self, function, *args, **kwargs):
+    def __init__(self, function=None, *args, **kwargs):  # pylint: disable=keyword-arg-before-vararg
```

Comment:
What is the reason for setting function to None as default? I expect, for example, that `executor.submit` will fail with an error if the default is kept.

Reply:
You're right, we don't actually need to default to `None` here, we only need to set `self._function = None` in `_prepare_dump` to allow serializing/deserializing. Otherwise, we'd run into `TypeError: cannot pickle '_thread.RLock' object`

Tested against my PR in qiskit-ibm-runtime and fixed in [a6d1314](https://github.com/Qiskit/qiskit/pull/12963/commits/a6d1314247cc4004e9a91777577822e998b40962)